### PR TITLE
Pure pursuit

### PR DESCRIPTION
**Pull request by RahulBansal1112**

### CODE REVIEW

File: src/main/java/frc/robot/LimeLight.java [hunk 1]
```diff
@@ -105,10 +104,10 @@ public void update() {
 	 * Outputs data to Shuffleboard.
 	 */
 	private void outputToShuffleboard() {
-		SmartDashboard.putBoolean("Has Target", hasValidTargets());
-		SmartDashboard.putNumber("Horizontal Offset", getXAngle());
-		SmartDashboard.putNumber("Vertical Offset", getYAngle());
-		SmartDashboard.putNumber("Area", getArea());
+		// SmartDashboard.putBoolean("Has Target", hasValidTargets());
```
Comment (AG6GR) [inline src/main/java/frc/robot/LimeLight.java]: Remove if no longer needed.

File: src/main/java/frc/robot/AutoSelector.java [hunk 2]
```diff
@@ -6,12 +6,26 @@
...
-		BLUE_3_BALL_B,
-		RED_1_BALL,
-		BLUE_1_BALL
+		RED_3_BALL(3),
+		BLUE_3_BALL(3),
+		RED_2_BALL(3),
+		BLUE_2_BALL(3),
+		RED_1_BALL(1),
+		BLUE_1_BALL(1),
+		LEAVE_TARMAC(0);
```
Comment (AG6GR) [inline src/main/java/frc/robot/AutoSelector.java:15]: Remove the LEAVE_TARMAC auto if it's not tested by end of day.

File: src/main/java/frc/robot/drive/Functions.java [hunk 3]
```diff
@@ -47,7 +47,15 @@ public static double calcForwardPower(double joystickInput) {
 	 * @return the adjusted steering power
 	 */
 	public static DrivePower calcSteeringPower(double steeringInput) {
-		return new DrivePower(1 + 2 * steeringInput, 1 - 2 * steeringInput);
+		if (steeringInput > 0) {
+			return new DrivePower(1 + 1 * Math.abs(steeringInput), 1 - 1 * Math.abs(steeringInput));
```
Comment (AG6GR) [inline src/main/java/frc/robot/drive/Functions.java]: Simplify expression, no need to abs() if you're adding the sign back in.

File: src/main/java/frc/robot/systems/DriveFSMSystem.java [hunk 4]
```diff
@@ -164,8 +182,8 @@ public void update(TeleopInput input) {
 
 		updateLineOdometry();
 		updateArcOdometry();
-		System.out.println("arc odo: " + robotPosArc.getX() + " " + robotPosArc.getY());
-		System.out.println("line odo: " + robotPosLine.getX() + " " + robotPosLine.getY());
+		// System.out.println("arc odo: " + robotPosArc.getX() + " " + robotPosArc.getY());
```
Comment (AG6GR) [inline src/main/java/frc/robot/systems/DriveFSMSystem.java]: Remove unused code.

File: src/main/java/frc/robot/systems/DriveFSMSystem.java [hunk 5]
```diff
@@ -416,8 +444,8 @@ private void handleForwardOrBackwardState(TeleopInput input,
 			setPowerForAllMotors(speed);
 		}
 
-		previousEncoderCount = currrentPosTicks;
-		System.out.println("Previous Encoder Count: " + previousEncoderCount);
+		previousEncoderCount = currentPosTicks;
+		// System.out.println("Previous Encoder Count: " + previousEncoderCount);
```
Comment (AG6GR) [inline src/main/java/frc/robot/systems/DriveFSMSystem.java]: Remove unused code

File: src/main/java/frc/robot/trajectory/AutoPaths.java [hunk 6]
```diff
@@ -84,13 +86,115 @@
...
+		path.add(new Translation2d(-225, -3));
+		return path;
+	}
+
+	/**
+	 * The blue 2 ball path.
+	 * @return an arraylist containing the key points
+	 */
+
+	public static ArrayList<Translation2d> b2BallPath() {
```
Comment (AG6GR) [inline src/main/java/frc/robot/trajectory/AutoPaths.java:125]: For Monterey: We should just have global public static ArrayList instances for these paths, not creating a new instance every time we look up a path.

File: src/main/java/frc/robot/systems/DriveFSMSystem.java [hunk 7]
```diff
@@ -563,21 +590,27 @@ private void handleTeleOpState(TeleopInput input) {
 				rightPower = turnPower;
 			}
 		}
-		System.out.println("dist: " + cvBallPos[0]);
-		System.out.println("angle: " + cvBallPos[1]);
+		// System.out.println("dist: " + cvBallPos[0]);
```
Comment (AG6GR) [inline src/main/java/frc/robot/systems/DriveFSMSystem.java]: Remove unused code

File: src/main/java/frc/robot/systems/DriveFSMSystem.java [hunk 8]
```diff
@@ -703,7 +754,153 @@ private void handleWaitToReceiveBallsState(TeleopInput input, double waitTime) {
 	 * @param path the path to run
 	 */
 	public void setAutoPath(DesiredMode path) {
-		autoPath = path;
+		switch (path) {
+			case RED_3_BALL:
+				startAngle = Constants.PP_R3_HUB_ANGLE_DEG;
```
Comment (AG6GR) [inline src/main/java/frc/robot/systems/DriveFSMSystem.java:726]: Would be nice to have a wrapper class around these auto parameters. I.e. here we'd do currentAuto = <some object>, and whenever we need to reference a value, we can just do currentAuto.target_angle or something like that.

**Review by AG6GR — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/AutoSelector.java (hunk 2 above)
Comment (RahulBansal1112) [reply src/main/java/frc/robot/AutoSelector.java:15]: Works.

**Review by AG6GR — DISMISSED** [review]

**Review by AG6GR — APPROVED** [review]

**Pull request merged**